src/inausoft.netCLI/Executor.cs

```diff
@@ -1,102 +1,139 @@
             }
 
+            var config = _config ?? _serviceProvider.GetService<Mapping>() ?? throw new InvalidOperationException();
+
             if (!args.Any())
             {
                 //TODO Add default handling here
-                throw new InvalidCommandException($"Command was not specified.");
+                return Fallback(ErrorCode.UnspecifiedCommand);
             }
 
-            var commandType = config._commandMap.Keys.FirstOrDefault(it =>
-                Attribute.IsDefined(it, typeof(CommandAttribute)) &&
-                (Attribute.GetCustomAttribute(it, typeof(CommandAttribute)) as CommandAttribute).Name == args[0]);
+            var mappingEntry = config.Entries.FirstOrDefault(it =>
+                 Attribute.IsDefined(it.CommandType, typeof(CommandAttribute)) &&
+                 (Attribute.GetCustomAttribute(it.CommandType, typeof(CommandAttribute)) as CommandAttribute).Name == args[0]);
 
-            if (commandType == null)
+            if (mappingEntry == null)
             {
-                throw new InvalidCommandException(args[0], $"Command {args[0]} was not mapped.");
+                return Fallback(ErrorCode.UnrecognizedCommand);
             }
 
-            var command = config.Deserializer.Deserialize(commandType, args.Skip(1).ToArray());
+            object command;
+
+            try
+            {
+                command = _deserializer.Deserialize(mappingEntry.CommandType, args.Skip(1).ToArray());
+            }
+            catch(DeserializationException ex)
+            {
+                return Fallback(ex.ErrorCode.Value);
+            }
+            catch (Exception)
+            {
+                return Fallback(ErrorCode.Unknown);
+            }
+            
 
-            var handler = serviceProvider.GetRequiredService(config._commandMap[command.GetType()]) as ICommandHandler;
+            var handler = (mappingEntry.HandlerInstance ?? _serviceProvider.GetRequiredService(mappingEntry.HandlerType)) as ICommandHandler;
 
             return handler.Run(command);
         }
 
-        public static int RunCLI(CLIConfiguration config, string[] args, params ICommandHandler[] handlers)
+        private int Fallback(ErrorCode errorCode)
         {
-            if (config == null)
-            {
-                throw new InvalidOperationException($"{nameof(CLIConfiguration)} was not registered. Run '{nameof(AddCLI)}' first.");
-            }
+            _fallbackFunc(errorCode);
 
-            if (!args.Any())
+            return (int)errorCode;
+        }
+
+        public static CLFlow Create()
+        {
+            return new CLFlow();
+        }
+    }
+
+    public static class netCLI
+    {
+        public static IServiceCollection ConfigureCLFlow(this IServiceCollection services, Action<Mapping> setup)
+        {
+            if (services == null)
```

Comment:
If this is an extension method, do you ever expect that `services` can be null? Not sure what is the common standard here!

Reply:
Could be. You might try to use it on uninitialized variable etc.